sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.formrecognizer;
+
+import com.azure.ai.formrecognizer.implementation.models.AnalyzeResult;
+import com.azure.ai.formrecognizer.implementation.models.DocumentResult;
+import com.azure.ai.formrecognizer.implementation.models.ReadResult;
+import com.azure.ai.formrecognizer.implementation.models.TextLine;
+import com.azure.ai.formrecognizer.implementation.models.TextWord;
+import com.azure.ai.formrecognizer.models.BoundingBox;
+import com.azure.ai.formrecognizer.models.DimensionUnit;
+import com.azure.ai.formrecognizer.models.Element;
+import com.azure.ai.formrecognizer.models.ExtractedReceipt;
+import com.azure.ai.formrecognizer.models.FieldValue;
+import com.azure.ai.formrecognizer.models.FloatValue;
+import com.azure.ai.formrecognizer.models.IntegerValue;
+import com.azure.ai.formrecognizer.models.LineElement;
+import com.azure.ai.formrecognizer.models.PageMetadata;
+import com.azure.ai.formrecognizer.models.Point;
+import com.azure.ai.formrecognizer.models.ReceiptItem;
+import com.azure.ai.formrecognizer.models.ReceiptType;
+import com.azure.ai.formrecognizer.models.StringValue;
+import com.azure.ai.formrecognizer.models.TextLanguage;
+import com.azure.ai.formrecognizer.models.WordElement;
+import com.azure.core.util.IterableStream;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * Helper class to convert service level models to SDK exposed models.
+ */
+final class Transforms {
+    private static final ClientLogger LOGGER = new ClientLogger(Transforms.class);
+    private static final Pattern COMPILE = Pattern.compile("[^0-9]+");
+
+    private Transforms() {
+    }
+
+    static IterableStream<ExtractedReceipt> toReceipt(AnalyzeResult analyzeResult, boolean includeTextDetails) {
+        List<ReadResult> readResults = analyzeResult.getReadResults();
+        List<DocumentResult> documentResult = analyzeResult.getDocumentResults();
+        List<ExtractedReceipt> extractedReceiptList = new ArrayList<>();
+
+        for (int i = 0; i < readResults.size(); i++) {
+            ReadResult readResultItem = readResults.get(i);
+            PageMetadata pageMetadata = getPageInfo(readResultItem);
+            ExtractedReceipt extractedReceiptItem = new ExtractedReceipt(pageMetadata);
+
+            DocumentResult documentResultItem = documentResult.get(i);
+            documentResultItem.getFields().forEach((key, fieldValue) -> {
+                switch (key) {
```

Comment:
Is it likely that there will be additional keys added in the future? If so, what is the plan to ensure this remains up to date when changes are made? Is there any possibility this could be made more generic and flexible to support additional keys?

Reply:
As this being a trained model, it would be unlikely to have additional keys or to expect this to be updated very often. Further, for any new introductions from the service side to keep this up to date this will have to go through updates accordingly.
Also, with such specific string matching, I am not sure how we could make it generic in nature?

Reply:
If these keys are not going to be updated frequently, can this be an ExpandableStringEnum?

Reply:
Update :
The service team intends to have additional keys added in between service updates. So to ensure this remains consistent the additional keys would be grouped into the `Map<String, FieldValue> extractedFields` [here]() and with proceeding client upgrades would be converted to strongly typed fields.

Reply:
Can the keys be converted to expandable enums?